### DISCUSSION TITLE
TSC Meeting November 2023

### DISCUSSION THREAD

**Discussion by dbaluta** [thesofproject/sof#8473] [Team Posts]
@thesofproject/steering-committee Lets talk about v2.8 release and planning new feature for v2.9

Register your availability: https://doodle.com/meeting/participate/id/e5WXRZZb

**Agenda** please add/edit items.

1) stable-v2.8 - release dates / features / status
2) v2.9 features and timeline.
3) Switching to native Zephyr interfaces - status
4) Timer domain vs DMA domain when switching to native Zephyr interface
5) IPC4 roadmap. All basic features upstream, plan on whats next and how to update.
6) Kernel roadmap, current plans and status of kernel work. Includes soundwire and SDCA readyness.
7) Zephyr roadmap, where we are today, what's next.
8) AOB
9) Release branch code maintenance and skipping main

**Comment by lgirdwood** [comment]
Fyi - @thesofproject/amd @thesofproject/google @thesofproject/mediatek @thesofproject/nxp @mwasko @plbossart please suggest topics and update availability. Thanks

**Comment by bhiregoudar** [comment]
Hi Liam, Updated the availability in doodle.

**Comment by lgirdwood** [comment]
@thesofproject/steering-committee fyi @dbaluta has sent out the invites for Tuesday 21st November at 4pm UK time.

**Comment by lgirdwood** [comment]
**Minutes**

@thesofproject/steering-committee pls find minutes and feel free to add more context or questions. Thank you all for attending.

1) v2.8

On track for code freeze for 24th November. A lot work around Zephyr refinement, Hifi4, Module API and DP scheduler introduction. Release target before end of December.

2) V2.9.

Timeline is to release before end of March 2024 with code freeze in 1st week of March. Features planned include deprecation of component API, header cleanups, modules, more HiFi and more native Zephyr. Linux topologies will also include DRC. 

3) Zephyr Native interfaces.

Some code like scheduler and memory APIs still using SOF legacy API that wrap Zephyr APIs, these will incrementally be changed to use more native Zephyr logic. Timer, DMA, DAIs all using native Zephyr APIs, but not fully in some use cases.

NXP using Hybrid Kconfig Zephyr driver mode for v2.8/v.9 whilst NXP drivers are transitioned. 

DMA scheduling domain no longer used by NXP. Could be deprecated in future. **Decision** Can only be deprecated when AMD/MTK also no longer need after move to Zephyr.


4) IPC3/4 roadmap. 

All basic IPC4 features upstream, plan on whats next and how to update. Changes can be made upstream in backwards compatible manner only.

Kconfig to select IPC3 or IPC4. NXP porting to IPC4, blocked at parse manifest header, Rimage tooling Intel specific for manifest generation, a cmd line switch could be used to select vendor data.

IPC3 future - Intel plans to use only IPC4 on new devices. NXP transitioning to IPC4, AMD/MTK currently using IPC3.

Other transport protocols. i.e. M4 cores using pmsg. Can use Zephyr infra, IPC4 can be embedded since messages are small.

5) Kernel roadmap

Intel/AMD to work together on parsing ACPI, Soundwire standard, need to support Intel/AMD vendor specific parts too.
 
DAIs all in same sound card - long term plan to split out HDMI on Intel deives,  but only when makes sense, i.e. AEC, not easy to split, orthogonal only.
 
Machine driver does not have ACPI/PCI to probe, SDW will eventually have way to probe machine driver, needs to be figured out with deferred and synchronous probes. ASoC plumbing required.

6) Zephyr roadmap.

Native APIs in lots of places now - still some subsystems need work like memory, scheduler.  Next steps - more native APIs and turn SOF into native Zephyr API user in 2024 with xtos APIs having the wrappers.

NXP plan to use hybrid driver mode until native driver mode ready. Intel did this with v2.3, v2.4, v2.5, AMD evaluating Zephyr.

7) Release branch code maintenance and skipping main

 Productions branches patches don't always end up in main.  Causes pain for Google, practice needs to be reduced. Every branch move causes large maintenance effort. AR: for IP vendors to internally improve process and be more like kernel with upstream 1st methodology.
 
 
 8) AOB
 
 @singalsu  to share plans for v2.9 topologies on return to office.
 
 WebRTC - Google has AEC, not in WebRTC. WebRTC had linking errors, old xtensa compilers had C++ errors, but now fixed with newer xtensa compilers and Zephyr C library support.  Speak to @andyross for C++ Zephyr linking.
